Apptainer dev

wellslu opened the pull request:
<!--
Thanks for contributing a pull request!
-->

#### Reference Issues/PRs
<!--
Example: "Fixes #1234", "See also #3456"
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

#### What does this implement/fix? Explain your changes.
<!--
Please include a brief explanation of how your solution
fixes the tagged issue(s), along with what files / entities have
been modified for this fix.
-->

add apptainer/singularity function

CH-chuan commented:
May I kindly ask why you closed the pull request? I think Apptainer env support is quite important, especially when you are running on certain clusters with no Docker permissions.

wellslu commented:
> May I kindly ask why you closed the pull request? I think Apptainer env support is quite important, especially when you are running on certain clusters with no Docker permissions.

Hi Chuan, thanks for your interest in my deployment. There are 2 reasons I close this pr. 
1, You are right, I added the Apptainer function into swe-agent because we are running our tests on clusters without root permission. But I am not the host of this project and the code is not completed yet. I just mis-sent this PR here lol. 
2, If you look into the code, you will see my apptainer function didn't completely merge into the original SWE-agent and SWE-ReX program structure. I added a new branch and some new files to only run with the Apptainer. I think it would ruin the author's original design, so I closed the PR. 
However, if you are still interested in my work, we absolutely can discuss it with each other :)
